kips/KIP01 - Prioritize Kolibri Liquidity Pool for Liquidations.md

```diff
@@ -0,0 +1,32 @@
+# Kolibri Improvement Proposal #01
+
+### Preface
+Hello everyone :)  
+I am honored to present the first Kolibri Improvement Proposal. I feel like Satoshi in some sense right now.  
+Kidding aside, I welcome and encourage everyone to participate in this and other proposals.  
+Lets not forget that as kDAO holders and as a community, we are in the control of the future and success of Kolibri.  
+I should also thank the Kolibri dev team for their hardwork and support.
+
+### Summary
+An onchain executable procedure to priorotize the Kolibri Liquidity Pool for oven liquidations.
+
+### Motivation
+The Kolibri Liquidity Pool (KLP) is an important and integral part of the kUSD algorithmic stable coin.  
+It serves at least the following purposes :
+1. Makes sure that big ovens that may not be liquidable by individuals, can be liquidated by the pooling the resources of individual kUSD investors.
+2. Provides a fair return for the investors who are willing to help the kUSD ecosystem stability. 
+
+
+The current implementation doesnt prioritize the KLP and individual investors compete for the liquidation of the under-collaterized ovens using high fees.  
+Such implementation is not healthy for the ecosystem and will discourage the pool investors and may lead to disintegration of KLP due to small or no returns of their investment into KLP.  
+In order to resolve this issue, I propose to give the KLP a X block priority for the liquidations.
+
+### Details
+In order to make an onchain executable procedure, I propose to make the Liquidation a 2 step process, Preliquidate and Liquidate.  
+If an oven gets under-collaterized and is not in Preliquidation state, The Preliquidate entrypoint should be called which sets the Preliquidation state and prioritization expiry block height in the oven storage.  
+The prioritization expiry block height is X blocks higher than the current block. (Can be 5 or 10 or ...)  
+If an oven is in Preliquidation state and block height is less than expiry block height, only the KLP can execute the liquidate entry point.  
```

Comment:
How do you deal with the case where (Assuming a buffer of 5 blocks):
- At block `n` an oven is liquidatable
- At block `n+1` the oven is marked for preliquidate
- At block `n+2` the oven is unliquidatable, but no one unmarks it
- At block `n+6` a private user liquidates the oven

Reply:
If an oven is in Preliquidate state, it doesnt mean that it is liquidatable. It only means that KLP has the priority to exercise Liquidate procedure for X blocks from the time that flag is set.
so at n+6 if the oven is still in Preliquidate state but not undercollaterized, no one can liquidate it anyway.

Reply:
I think we're over-complicating this a bit, why not just make the LP able to liquidate as long as ovens are under-collateralized (same with the stability fund), and then manage the individual actor path as its own case (with the block level tracking)?